addons/pos_loyalty/static/src/overrides/models/pos_store.js

```diff
@@ -96,16 +96,17 @@ patch(PosStore.prototype, {
             if (trimmedCode && trimmedCode.startsWith("044")) {
                 // check if the code exist in the database
                 // if so, use its balance, otherwise, use the unit price of the gift card product
-                const fetchedGiftCard = await this.orm.searchRead(
+                const fetchedGiftCard = await this.data.searchRead(
                     "loyalty.card",
                     [
                         ["code", "=", trimmedCode],
                         ["program_id", "=", program.id],
                     ],
                     ["points", "source_pos_order_id"]
                 );
+
                 // There should be maximum one gift card for a given code.
-                const giftCard = fetchedGiftCard[0];
+                const giftCard = fetchedGiftCard["loyalty.card"][0];
```

Comment:
```suggestion
                const giftCard = fetchedGiftCard[0];
```
`searchRead` is supposed to return an array of records, right?

Reply:
Nop because of loadDate method on related_model